cfps/CFP-041-clang-format-for-bpf-formatting.md

```diff
@@ -0,0 +1,81 @@
+
+code formatting / code style is infamous for being a topic that can lead to heated debates and bikeshedding. Unfortunately chances are slim that we can make every contributor happy. We propose to create a small committee we trust to make these decisions and to given them the power to make these decisions. The committee will consist of Dylan Reimerink being the committer to work on this primarily and two other people yet to be chosen.
+
+### Existing format and `.clang-format` config
+
+The starting point of this work will be the existing format. We already have a `.clang-format` config file in the repository which has been created in the past to approximate the existing formatting style. The committee will review the changes made by clang-format and decide on the best settings to use.
+
+### Initial re-formatting commit
+
+file so the commit doesn't mess up the blame history.codebase and commit the changes. This will be a single commit to avoid polluting the git history with formatting changes. We will add this commit to a `.git-blame-ignore` which is recognized by both the git cli tool and GitHub to ignore this commit when showing blame information to minimize the impact.
+
+### Tooling changes
+
+We propose to update the cilium-builder docker image which builds our clang to also provide `clang-format`. We will add make targets to run `clang-format` in-place (directly update files) or output any suggested changes to stdout.
+
+In CI we will run one of these targets to detect if a committer properly formatted their code (clang-format should give no suggestions).
+
+We will modify the checkpatch script in the checkpatch docker iamge, disabling the formatting checks now covered by clang-format and leaving all other checks enabled.
+
+### Backporting
+
+We propose to backport the config, tooling and CI changes to older branches we still maintain. Then run clang-format on these branches to update their formatting in the same way and adding the commit in a `.git-blame-ignore` file.
+
+Causing the older branches to be re-formatted in the same way should minimize merge conflicts when backporting changes. Since formatting only re-arranges code and doesn't change the code itself, the risk of doing this should be minimal.
+
+### Bypassing the normal CODEOWNERS review process
+
+Finally, we propose to bypass the normal CODEOWNERS review process when merging. Instead, the committee should review and approve the PR(s) needed for this change. Otherwise, we would involve additional people at the very end of the process making the purpose of the committee pointless.
```

Comment:
Add a rationale why we need to bypass the usual process? It would also be good to put in here who is going to toggle the "Bypass branch protection" tickbox to get this merged. tophat? Andre / Joe?

Reply:
> Add a rationale why we need to bypass the usual process?

I thought that was the last line of this paragraph, do you have a better way of wording this or a different rationale?

> who is going to toggle the "Bypass branch protection" tickbox to get this merged. tophat? Andre / Joe?

Yes, but I don't know at this moment. Logically it would have to be "A Cilium admin" or the Tophat. I had hoped someone would step up if brought up in discussion. We can only ask and not appoint someone

Reply:
Okay, makes sense. I would still explicitly mention that this will require bypassing branch protection.